crates/fuel-core/src/graphql_api/storage/balances.rs

```diff
@@ -0,0 +1,76 @@
+use fuel_core_storage::{
+    blueprint::plain::Plain,
+    codec::{
+        postcard::Postcard,
+        raw::Raw,
+    },
+    structured_storage::TableWithBlueprint,
+    Mappable,
+};
+use fuel_core_types::{
+    fuel_tx::{
+        Address,
+        AssetId,
+    },
+    fuel_vm::double_key,
+};
+use rand::{
+    distributions::Standard,
+    prelude::Distribution,
+    Rng,
+};
+
+pub type Amount = u64;
+
+double_key!(BalancesKey, Address, address, AssetId, asset_id);
+impl Distribution<BalancesKey> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> BalancesKey {
+        let mut bytes = [0u8; BalancesKey::LEN];
+        rng.fill_bytes(bytes.as_mut());
+        BalancesKey::from_array(bytes)
+    }
+}
+
+impl core::fmt::Display for BalancesKey {
```

Comment:
Not for this PR, but is there any benefit in putting the `Display` implementation inside the `double_key` macro?

Reply:
I was considering this, but after giving this a second thought I came to the conclusion that `Display` implementation is often pretty specific to a particular use-case.

For example, if one part is the "private key" I may want to display `*****` in the log. If `Display` is automatically provided, this may go unnoticed.